banzai_floyds/matched_filter.py

```diff
@@ -0,0 +1,315 @@
+    filter signal.
+
+    Parameters
+    ----------
+    error: array of uncertainties, should be the same shape as the input data
+    weights: array of match filter weights, should be the same shape as the input data
+
+    Returns
+    -------
+    float: normalization value for the matched filter metric
+
+    Notes
+    -----
+    Note that the normalization includes the square of the weights. The propagation of uncertainty for a weighted
+    sum of independent measurements is variance = Σ (w / σ²)² σ² =  Σ w² / σ²
+
+    """
+    return ((weights * weights / error / error).sum()) ** 0.5
+
+
+def matched_filter_metric(theta, data, error, weights_function, weights_jacobian_function,
+                          weights_hessian_function, x, *args):
+    """
+    Calculate the matched filter metric to optimize your model. This is the matched filter signal / the sqrt of
+    the variance of the matched filter signal (S/N)
+
+    Parameters
+    ----------
+    theta: array of input values for the parameters of the weights function that in principle can be varied
+           using scipy.optimize.minimize
+    data: array of the data to match filter
+    error: array of uncertainties, should be the same size as the data array
+    weights_function: callable function to calculate the match filter weights. Should return an array the same shape as
+                     input data.
+    weights_jacobian_function: Not used in this function. This is only included here because scipy.minimize.optimize
+                               passes the same arguments to jacobian and hessian functions.
+    weights_hessian_function: Not used in this function. This is only included here because scipy.minimize.optimize
+                              passes the same arguments to jacobian and hessian functions.
+    x: tuple of arrays independent variables x, y. Arrays should be the same shape as the input data
+    args: tuple of any other static arguments that should be passed to the weights function.
+
+    Returns
+    -------
+    float: signal-to-noise metric for the matched filter
+    """
+    weights = weights_function(theta, x, *args)
+    metric = matched_filter_signal(data, error, weights)
+    metric /= matched_filter_normalization(error, weights)
+    return metric
+
+
+def matched_filter_signal_jacobian(theta, x, data, error, weights_jacobian_function, *args):
+    """
+    Calculate the jacobian of the numerator of the match filter metric
+
+    Parameters
+    ----------
+    theta: array of input values for the parameters of the weights function that in principle can be varied
+           using scipy.optimize.minimize
+    x: tuple of arrays independent variables x, y. Arrays should be the same shape as the input data
+    data: array of the data to match filter
+    error: array of uncertainties, should be the same size as the data array    x
```

Comment:
extra tab here?

Reply:
When the description is a second line for the same parameter, don't you add a visual indent? I'm looking at things like this: https://docs.astropy.org/en/stable/_modules/astropy/io/fits/convenience.html#writeto
I guess it is only a single indent?